server/src/main/java/org/apache/uniffle/server/storage/MultiStorageManager.java

```diff
@@ -65,39 +73,51 @@ public Storage selectStorage(ShuffleDataReadEvent event) {
 
   @Override
   public void updateWriteMetrics(ShuffleDataFlushEvent event, long writeTime) {
-    selectStorageManager(event).updateWriteMetrics(event, writeTime);
+    throw new UnsupportedOperationException();
   }
 
   @Override
-  public boolean write(Storage storage, ShuffleWriteHandler handler, ShuffleDataFlushEvent event) {
+  public boolean write(Storage storage, ShuffleWriteHandler handler, ShuffleDataFlushEvent event, 
+                       CreateShuffleWriteHandlerRequest request) {
```

Comment:
Why do we add the parameter `request`? Why don't use the origin way?

Reply:
Because `ShuffleWriteHandler` is create by old `Storage`, we need to create a new handler with `request` after fallback. I found out a bug here just now.

Reply:
In origin way, we can get the request from the storage.

Reply:
Do you mean we can get the request from the old storage when create a new handler?

Reply:
> Because `ShuffleWriteHandler` is create by old `Storage`, we need to create a new handler with `request` after fallback. I found out a bug here just now.

I found a bug in `org.apache.uniffle.storage.common.AbstractStorage#containsWriteHandler` and i will create another pr to solved

Reply:
> Do you mean we can get the request from the old storage when create a new handler?

Yes.